### DISCUSSION TITLE
Add Stealth Music to BeardLib

### DISCUSSION THREAD

**Discussion by ToughSun14** [diesel-modding/PAYDAY-2-BeardLib#647] [Ideas]
Not sure if anyone else has suggested this, but I'll throw my hat in the ring in case it hasn't. So, I recently migrated over to the CustomOST Fixer mod after the original was causing crashes for me, and while I do appreciate it for not crashing my game (so far), one thing I noticed is that none of the tracks I created for stealth-only heists (shadow raid, murky station, etc.) show up in-game. I would really love it if Stealth tracks support was added. I mean the [Custom OST](https://modworkshop.net/mod/27359) mod was able to do it, so why not add it to BeardLib?

**Comment by segabl** [comment]
Stealth music is already a thing, though looking at the wiki, seems to be not documented anywhere yet. instead of `HeistMusic` you just write `StealthMusic` and instead of the assault events you use events suspense_1, suspense_2, suspense_3 and suspense_4. I'll keep this open though cause that definitely should be added to the wiki.

**Reply by ToughSun14** [reply]
Okay, well while we're at it, does the alt file feature from CustomOST also exist in the BeardLib framework? From what I saw of the documentation, it's apparently been deprecated, but I think I noticed it working with the CustomOST Fixer mod. So is it deprecated, or has its functionality not been properly updated in the documents?

**Reply by segabl** [reply]
no, just that specific way of defining alt tracks is deprecated. It still works but it's not guaranteed to keep working and won't receive any changes. Further down on the wiki page there's an example how to properly do alt tracks.

**Reply by ToughSun14** [reply]
1) You sure? I don't exactly see anywhere on the heist music page about how to add alt tracks.
2) Is it possible to have both heist and stealth tracks in one music mod?

**Reply by segabl** [reply]
1. 
![image](https://github.com/simon-wh/PAYDAY-2-BeardLib/assets/4538631/9b4b3e73-c3a0-4244-8724-52d8d9bb063c)

2.
yes you can have as many modules in a main.xml as you want, so you can have both a heistmusic and a stealthmusic node or even multiple.

**Reply by ToughSun14** [reply]
Oh, okay. Well, while we're talking about music, got a bug to report: if one of the songs made with BeardLib's framework is playing, there are certain circumstances where it won't stop playing. Situation 1: I was browsing through the custom songs choosing one to play, and suddenly the heist starts. Result: The music doesn't stop until the opening cutscene ends and then the setup track plays. Situation 2: I suddenly decide to leave a heist. Result: the current music doesn't stop playing until it ends naturally and THEN the menu music starts.

**Reply by ToughSun14** [reply]
Also, I've been working on creating a new track using the BeardLib base, but no matter what folder I put it in, it doesn't work. Where exactly do I place the mod once it's done? Also, in case I already am putting it in the right folder, do you think you could take a look at my music mod and tell me what I'm doing wrong, if anything?
[ScarfaceMusic.zip](https://github.com/simon-wh/PAYDAY-2-BeardLib/files/11470510/ScarfaceMusic.zip)

**Reply by segabl** [reply]
every track node needs an audio source, a start_source is not supposed to be on its own track node, it goes onto the node that you want it to be the "intro" music for. Your mod probably doesnt work because of that, check the BLT log in mods/logs for errors.

**Reply by ToughSun14** [reply]
1) Well, I tried your advice, and it worked like a charm. Thanks!
2) Well, I tried setting up a stealth track, but it didn't seem to work correctly. The game read the files and everything since it showed up in-game as a track I could play, but none of the sounds from the track played. Think you could take a look and see what I did wrong?
[WaitForGauntlet.zip](https://github.com/simon-wh/PAYDAY-2-BeardLib/files/11472145/WaitForGauntlet.zip)

**Reply by segabl** [reply]
did you use it on a stealth only heist like shadow raid or yacht? afaik it only works there.

**Reply by ToughSun14** [reply]
Yes. I even tried testing it out in the playlist settings menu, but no sound played.

**Reply by ToughSun14** [reply]
Also, noticed another glitch with the BeardLib music system: If I change music tracks in the lobby, then the regular lobby music doesn't start playing again after I leave the soundtrack tab.

**Reply by ToughSun14** [reply]
Well? Any updates on fixes to the Stealth Music Module?
(before you ask, yes, I am aware the wiki was updated with proper documentation, and I followed the instructions, but the sound is STILL not playing)

**Reply by segabl** [reply]
can you attach the mod you are trying to make so I can check what might be going wrong?

**Reply by ToughSun14** [reply]
Here
[stealth_waitforgauntlet.zip](https://github.com/simon-wh/PAYDAY-2-BeardLib/files/12336158/stealth_waitforgauntlet.zip)

**Reply by segabl** [reply]
not sure what's wrong, it works fine for me, other than the localization error. started shadow raid, set this track as heist track and it played just fine

**Reply by ToughSun14** [reply]
1) Wait, there's a localization error? Where?
2) Really? You got it working, but I couldn't? What's your setup for the files? I want to see if what you do works for me, and how it's different from my setup.

**Reply by ToughSun14** [reply]
Actually, a thought occurs. Is it possible that there could be another mod interfering with BeardLib's module? And if so, what might it be?

**Reply by segabl** [reply]
when you select the track in the menu its name is "Error: blabla..." so something is wrong with the localization. You should probably check the BLT log to see if something interferes with your mod. I simply placed it into mod_overrides and it worked just fine.

**Reply by ToughSun14** [reply]
Okay... and where exactly IS the BeardLib log located? (no, seriously, I have no goddamn idea where it is.)

**Reply by ToughSun14** [reply]
(also, the reason you had an error for the music name was because I forgot to include the "loc" folder in the zip)

**Reply by segabl** [reply]
mods/logs

**Reply by ToughSun14** [reply]
Okay. Well, I gave it another shot yesterday, and once again, the BeardLib stealth music didn't play. I honestly have no idea what I might be doing wrong. Is it possible that, in some way, Custom OST is interfering with BeardLib's music module?
